### DISCUSSION TITLE
بحث پیرامون نکات ترجمهٔ فارسی

### DISCUSSION THREAD

**Discussion by mahdyar** [mahdyar/fa.javascript.info#1] [Ideas]
اینوایت شدی. @mahdihashemi14

**Comment by mahdiHash** [comment]
مرسی

**Comment by mahdiHash** [comment]
الان ما باید داخل این ریپازیتوری نکات رو بنویسیم شما داخل ریپوی اصلی مرج کنین درسته؟

**Reply by mahdyar** [reply]
> الان ما باید داخل این ریپازیتوری نکات رو بنویسیم شما داخل ریپوی اصلی مرج کنین درسته؟

آره.

**Reply by mahdiHash** [reply]
اوکی من هر وقت بتونم شروع میکنم.

با چه چیزی شروع کنیم؟ با کلمات یا نکات ترجمه؟

این نکاتی که پیش‌فرض هستن رو هم ترجمه کنیم یا بزاریم همونطوری بمونن؟

**Reply by mahdyar** [reply]
به نظرم یک بخش اضافه کنیم زیر همون translation tips به فارسی بنویسیم نکات ترجمهٔ فارسی.
بقیه‌ش همینطوری بمونه.
@mahdiHashemi14

**Reply by mahdiHash** [reply]
فکر خوبیه👍🏻

**Comment by mahdiHash** [comment]
یه نکته هم میخوام درباره ترجمه ارورهای تو کامنت‌ها اضافه کنم
من خودم ارورهایی که بدون تغییر نوشته میشن رو ترجمه نمیکنم که اگه کاربر کد رو اجرا کرد همون پیام رو ببینه
ولی بعضی از ارورها دقیقا همونی که مرورگر نشون میده نیست، یعنی یه جورایی ارور رو به صورت کوتاه نوشته اونجا. این ارورها رو من ترجمه میکنم
داخل نکات هم همینجوری میخوام بنویسم
خوبه؟

**Reply by mahdyar** [reply]
> یه نکته هم میخوام درباره ترجمه ارورهای تو کامنت‌ها اضافه کنم من خودم ارورهایی که بدون تغییر نوشته میشن رو ترجمه نمیکنم که اگه کاربر کد رو اجرا کرد همون پیام رو ببینه ولی بعضی از ارورها دقیقا همونی که مرورگر نشون میده نیست، یعنی یه جورایی ارور رو به صورت کوتاه نوشته اونجا. این ارورها رو من ترجمه میکنم داخل نکات هم همینجوری میخوام بنویسم خوبه؟

آره اوکیه.

**Comment by mahdiHash** [comment]
سلام.
موافقی تیترهای داخل readme رو یه جوری متفاوت‌تر کنیم؟
الان خیلی ساده‌س شبیه متن معمولیه اگه بخوایم دنبال تیترها بگردیم یه ذره سخت میشه.
مثلا اگه پایین تیتر "---" بزاریم یه خط زیرش میاد ولی یه مقدار زشت میشه. یا مثلا اگه قبل از "###" یه دونه "-" بزاریم یه دایره میاد بغل تیتر.

**Reply by mahdyar** [reply]
[این](https://github.blog/changelog/2021-04-13-table-of-contents-support-in-markdown-files/) کافی نیست؟

**Reply by mahdiHash** [reply]
اره خوبه👍🏻

**Reply by mahdiHash** [reply]
😅ولی تو چشم نیست. امیدوارم بقیه بدونن اونجاست

**Reply by mahdyar** [reply]
> 😅ولی تو چشم نیست. امیدوارم بقیه بدونن اونجاست

می‌تونیم خودمون هم یک TOC جنریت کنیم. 🤔

**Reply by mahdiHash** [reply]
> > 😅ولی تو چشم نیست. امیدوارم بقیه بدونن اونجاست
> 
> می‌تونیم خودمون هم یک TOC جنریت کنیم. 🤔

بنظرم تیترها انقدر زیاد نیستن که خودمون یه دونه بسازیم. پایین هر کدوم خط بذاریم کافیه.

می‌تونیم امتحان کنیم اگه بد شد برمی‌گردونیم

**Reply by mahdyar** [reply]
> > > 😅ولی تو چشم نیست. امیدوارم بقیه بدونن اونجاست
> > 
> > 
> > می‌تونیم خودمون هم یک TOC جنریت کنیم. 🤔
> 
> بنظرم تیترها انقدر زیاد نیستن که خودمون یه دونه بسازیم. پایین هر کدوم خط بذاریم کافیه.
> 
> می‌تونیم امتحان کنیم اگه بد شد برمی‌گردونیم

اوکیه.

**Comment by mahdiHash** [comment]
بنظرت ترجمه decorator چی میشه؟
الان معنی‌ش رو نوشتم دکوراتور.
تو این [مقاله](https://github.com/javascript-tutorial/fa.javascript.info/blob/master/1-js/06-advanced-functions/09-call-apply-decorators/article.md) استفاده میشه

**Reply by mahdyar** [reply]
> بنظرت ترجمه decorator چی میشه؟ الان معنی‌ش رو نوشتم دکوراتور. تو این [مقاله](https://github.com/javascript-tutorial/fa.javascript.info/blob/master/1-js/06-advanced-functions/09-call-apply-decorators/article.md) استفاده میشه

اوکیه همون. یک معادل فارسی اگر براش پیدا کنیم ممکنه باعث سردرگمی بشه.
 تو گوگل هم سرچ کردم یا همون Decorator انگلیسی نوشتن یا دکوراتور نوشتن. :)

**Comment by mahdiHash** [comment]
سلام.  
زیر تیترها داخل readme خط گذاشتم.  
ببین خوب شده؟ اگه نه برگردونیم به حالت قبلی

**Reply by mahdyar** [reply]
> سلام. زیر تیترها داخل readme خط گذاشتم. ببین خوب شده؟ اگه نه برگردونیم به حالت قبلی

سلام
نه جالب نشده بود، revert کردم.

**Comment by mahdyar** [comment]
@mahdiHashemi14 این لغت‌نامه واقعاً نیازه؟ یکم داره زیادی طولانی می‌شه readme، و معمولاً چیزی که زیادی طولانی شه رو نمی‌خونن.

**Reply by mahdiHash** [reply]
<div dir="rtl">

لغت‌نامه رو اخرش گذاشتم که نکات قبلی رو بخونن و فقط موقع نیاز برن سراغ لغت‌نامه.  
بنظرم وجود داشته باشه بهتره چون ترجمه کلمات یک‌دست میشه.
مثلا کلمه iterable رو من حلقه‌پذیر داخل مقاله ترجمه کرده بودم ولی فکرکنم امیرحسین ویسی یه چیز دیگه ترجمه کرده بود. اگه ترجمه واحد وجود باشه خواننده راحت‌تر متوجه میشه
فکرکنم برنچ‌های دیگه هم لغت‌نامه داشته باشن. ولی اونا چون اکثرا از چپ به راست می‌نویسن نیازی به نکته ترجمه کامنت و رشته و اینا ندارن. برای همین readme برنچ ما داره خیلی طولانی میشه
</div>

**Reply by mahdyar** [reply]
جز اون، ریویو رو چی‌کار کنم؟ :) اینا پیشنهاد باشن مثلاً؟ چون نمی‌شه برای هر PR بشینم دونه دونه کلمه‌ها رو ببینم چی ترجمه کردن توی لغت‌نامه چی بوده...

**Reply by mahdiHash** [reply]
اینا پیشنهاد نیستن ولی خب توقع‌مون اینه که مترجم اگه ترجمه یه کلمه رو داخل جدول ببینه همون رو استفاده کنه.
اینکه مترجم از اینا استفاده کنه یا نه دیگه به خودش بستگی داره. اگه شما دیدی یه کلمه اشتباهه بهش بگو ولی نیازی نیست تک تک همه کلمات رو چک کنی؛ هم سخته هم وقت‌گیر😅

**Comment by mahdyar** [comment]
@mahdiHashemi14 
این الآن تمومه؟ چیز دیگه‌ای مونده؟ اگر مونده لیست کن منم چندتاشو اضافه کنم.

**Reply by mahdiHash** [reply]
تا حالا کلمه‌هایی که به ذهنم رسید یا داخل ترجمه دیدم رو اضافه کردم.
بعدا شاید نیاز بشه بازم کلمه اضافه کنیم.
ولی نکته‌ها تموم شدن.
شما اگه چیزی می‌خوای اضافه کن.

**Comment by mahdiHash** [comment]
&rlm; چون الان readme طولانی شده، بنظرم اولش بنویسیم لغت‌نامه آخرین بخشه که اگه یه نفر خواست بخونه فکر نکنه این حجم کلا نکته‌ست &lrm;
